Source/WebCore/PAL/pal/PALSwift/UnsafeOverlays.swift

```diff
@@ -30,12 +30,92 @@ import CryptoKit
 
 import PALSwift
 
+enum UnsafeErrors: Error {
+    case invalidLength
+    case emptySpan
+}
+
 extension HashFunction {
     mutating func update(data: SpanConstUInt8) {
-        self.update(
-            bufferPointer: UnsafeRawBufferPointer(
-                start: data.__dataUnsafe(), count: data.size()))
+        if data.empty() {
+            self.update(data: Zero.zeroData)
+        } else {
+            self.update(
+                bufferPointer: UnsafeRawBufferPointer(
+                    start: data.__dataUnsafe(), count: data.size()))
+        }
+    }
+}
+
+// Ugly but needed.
+// all __dataUnsafe() calls here return nil if input WTF::Vector is zero length.
+// CommonCrypto APIs handled nil pointers internally. Swift APIs crash.
+// To maintain backward compatibility, a zero array is needed to go past Swift Validation
+// and let CyptoKit replicate behavior.
+private struct Zero {
+    public static var zeroData = Data([UInt8]())
+}
+
+extension ContiguousBytes {
+    public func createVectorUInt8() -> VectorUInt8 {
+        return self.withUnsafeBytes { buf in
+            var result = VectorUInt8(buf.count)
```

Comment:
This function achieves our goal. But it is forced to name the number of bytes four times, and technically copyBytes skips the bounds check. It would be slightly nicer if you could make an explicit "makeSpan" or "makeSpanUInt8" function in this file, and then we'd have just one line of code:

`var result = VectorUInt8(makeSpan(buf.baseAddress, buf.count))`